northstar-tests/src/logger.rs

```diff
@@ -14,75 +14,81 @@
 
 use anyhow::{anyhow, Context, Result};
 use lazy_static::lazy_static;
-use log::{debug, warn};
+use log::debug;
 use regex::Regex;
 use std::{
     fmt,
     io::Write,
     time::{Duration, Instant},
 };
-use tokio::{pin, select, time};
+use tokio::{
+    sync::{
+        mpsc::{unbounded_channel, UnboundedReceiver, UnboundedSender},
+        Mutex,
+    },
+    time,
+};
 
 lazy_static! {
+    /// Test started timestamp
     static ref START: Instant = Instant::now();
-    static ref QUEUE: (flume::Sender<String>, flume::Receiver<String>) = flume::unbounded();
+    /// Channel for log assumptions
+    static ref QUEUE: (UnboundedSender<String>, Mutex<UnboundedReceiver<String>>) = {
```

Comment:
interesting pattern...did you use a mutex here just to get rid of the `flume` dependencies?
is this preferable to using a broadcast channel with subscriptions?

Reply:
The `Mutex` is for getting a `&mut` to the receiver because of [recv](https://docs.rs/tokio/1.9.0/tokio/sync/mpsc/struct.Receiver.html#method.recv).

A `BroadcastReceiver` would require a `subscribe` in *every* `assume` call. The resulting stream would lack all logs *before* the `subscribe` so this would not work.

The `broadcast` is a thing "to be avoided" if possible because you have to handle the `Lagged` errors properly and that's not easy.

Reply:
ok thanks